### DISCUSSION TITLE
Bookmarked connection fails

### DISCUSSION THREAD

**Discussion by blue-genie** [electerm/electerm#3178] [Q&A] [answered]
Hello,

I have a strange issue. 
If I so `ssh ec2-user@XX.XX.XX.XX -i ~/.../.../test.pem` the SSH connection is established. 
If I create a bookmark(saved connection) and provide the same information, that connection fails.

Any guidance on what I could do?

**Comment by blue-genie** [comment]
Looking further I see that the server logs:
`key type ssh-rsa not in PubkeyAcceptedAlgorithms`

But if I sent the same key from command line, it works.

**Comment by blue-genie** [comment]
`ssh-keygen -l -f ~/.../.../test.pem` returns `2048 SHA256:/YqOC0i31B8195ftXUWpR1Nr6iXXXXP27O2O58 no comment (RSA)` - so it's already SHA2

**Comment by zxdong262** [comment]
How is the pem created? electerm do have limited support for private key format, if it is ceated with ssh-keygen cmd, should be OK, other than that, might not be supported, this is something should be improved,  I will check what I can do, 

also might be related to: https://github.com/mscdex/ssh2/issues/1316 , if that is the case, use ed25519 keyformat may work, or maybe edit .ssh/config, add:

```
addKeysToAgent yes
PubkeyAcceptedKeyTypes=+ssh-rsa
```

**Reply by blue-genie** [reply]
I don't remember how it was generated, it was a while ago. The format is:

```
-----BEGIN RSA PRIVATE KEY----
...
-----END RSA PRIVATE KEY-----
```

but the same key works if it's used to another another using electerm. 

I'm guessing that on the other server, `ssh-rsa` is in `PubkeyAcceptedAlgorithms` but on this server it's not.

Does electerm, or the libraries that it uses, convert the file or something?

Why does it work if I use ssh command inside electerm?

**Reply by blue-genie** [reply]
AWS blocks `ssh-rsa` for a reason' I don't want to expose my server to security issues.

**Reply by zxdong262** [reply]
Do you permission to create new key pairs In ec2 dashboard?
If you can create key pairs, can you try create the ed25519 format key pair to test?

**Comment by zxdong262** [comment]
Can you create a temp ec2 key pair for me to test, the private key and pubkey if could get the pubkey.

**Reply by zxdong262** [reply]
no need any more, after updated ssh2 module, as I tested, should be able to connect with rsa key, so would fix in next release

**Comment by zxdong262** [comment, marked as answer]
Can you try https://github.com/electerm/electerm/releases/tag/v1.37.80

**Reply by blue-genie** [reply]
I'm sorry for the delay, I was side tracked.

I upgraded to 1.37.80 - all is good. Thank you

**Reply by zxdong262** [reply]
Thank you for the feedback~or I may not resolve it.
